### PR TITLE
Fix: argent mobile connection

### LINKED ISSUE #915: Argent Mobile not working
## Description 📹

The Argent Mobile connection doesn't work on the Starknet ID website atm we have

telegram-cloud-photo-size-4-5882024735292900841-y

## Proposed Actions 🛠️

Here’s a checklist of actions to follow for resolving this issue:

1. **Fork and Create Branch**:  
   Fork the repository and create a new branch using the issue number:
   ```bash
   git checkout -b fix-[issue-number]
   ```

2. **Implement Changes**:  

- [ ] Understand why no connectors are showing and fix it (make sure to test it's working with the build after pushing)

3. **Run Tests and Commit Changes**:  
   Make sure your changes don't break existing functionality and commit with a clear message:
   ```bash
   git commit -m "Fix: [Short description of the fix]"
   ```

## Required 📋

To keep our workflow smooth, please make sure you follow these guidelines:

- **Assignment**: Don't create a pull request if you weren’t assigned to this issue.
- **Timeframe**: Complete the task within **3 business days**.
- **Closing the Issue**: In your PR description, close the issue by writing `Close #[issue_id]`.
- **Review Process**:
  - Once you've submitted your PR, change the label to **"ready for review"**.
  - If changes are requested, address them and then update the label back to **"ready for review"** once done.
- **Testing**: Test your PR locally before pushing, and verify that tests and build are working after pushing.

Thank you for your contribution 🙏

⚠️ WARNING: Failure to follow the requirements above may result in being added to the OnlyDust blacklist, affecting your ability to receive future rewards.

### PR DESCRIPTION
**Pull request by JoE11-y**
fixes #915 

Changes made:

- Updated the argentinbrowser check to use the provided method by the starknetkit sdk

## Summary by CodeRabbit

- **New Features**
	- Introduced new environment variables for connecting to the StarkNet Sepolia network and application ID.
- **Bug Fixes**
	- Simplified the `WalletConnect` component by enhancing connector filtering logic and removing unnecessary state management.
- **Chores**
	- Updated `.gitignore` to exclude `.env` files from version control.
	- Removed unused imports and a specific function to streamline the codebase.

### CODE REVIEW

File: .env.test [hunk 1]
```diff
@@ -2,3 +2,4 @@ NEXT_PUBLIC_VERIFIER_CONTRACT=0x057c942544063c3aea6ea6c37009cc9d1beacd750cb68015
 NEXT_PUBLIC_DEPRECATED_VERIFIER_CONTRACT=0x019e5204152a72891bf8cd0bed8f03593fdb29ceacd14fca587be5d9fcf87c0e
 NEXT_PUBLIC_OLD_VERIFIER_CONTRACT=0x4d546c8d60cfd591557ac0613be5ceeb0ea6f797e7d11c0b5160d145fa3089f
 NEXT_PUBLIC_VERIFIER_POP_CONTRACT=0x03528caf090179e337931ee669a5b0214041e1bae30d460ff07d2cea2c7a9106
+NEXT_PUBLIC_RPC_URL=https://starknet-sepolia.infura.io/v3/2E9u9FUJucyGntaYzan2LNJdEGK
```
Comment (fricoben) [inline .env.test]: delete but you can add `NEXT_PUBLIC_RPC_URL=https://sepolia.rpc.starknet.id`

File: components/UI/walletConnect.tsx [hunk 2]
```diff
@@ -102,13 +101,13 @@ const WalletConnect: FunctionComponent<WalletConnectProps> = ({
...
                   <img
-                    src={getConnectorIcon(connector.id)}
+                    src={isInArgentMobileAppBrowser() ? getConnectorIcon('argentMobile') : getConnectorIcon(connector.id)}
                     className={styles.walletIcon}
                   />
                   <div className={styles.walletName}>
                     <p>
                       {needInstall(connector, isAvailable) ? "Install " : ""}
-                      {connector.id === "argentMobile" && isMobile
+                      {connector.id === "argentMobile" && isMobile || isInArgentMobileAppBrowser()
```
Comment (fricoben) [inline components/UI/walletConnect.tsx]: connector.id === "argentMobile" && isMobile 

seems similar to 

isInArgentMobileAppBrowser()

Wouldn't just `isInArgentMobileAppBrowser()` work here ?

File: components/UI/walletConnect.tsx [hunk 3]
```diff
@@ -8,9 +8,9 @@ import {
   getConnectorDiscovery,
   getConnectorIcon,
   getConnectorName,
-  isInArgentMobileAppBrowser,
   sortConnectors,
 } from "@/utils/connectorWrapper";
+import { isInArgentMobileAppBrowser } from 'starknetkit/argentMobile';
```
Comment (fricoben) [inline components/UI/walletConnect.tsx:13]: Nice didn't know it existed

File: components/UI/walletConnect.tsx [hunk 4]
```diff
@@ -26,21 +26,20 @@ const WalletConnect: FunctionComponent<WalletConnectProps> = ({
...
 
-  useEffect(() => {
-    if (typeof window !== "undefined")
-      setIsArgentMobile(isInArgentMobileAppBrowser());
-  }, []);
-
   const filterConnectors = (connectors: Connector[]) => {
-    if (isArgentMobile) {
-      return connectors.filter((connector) => connector.id === "argentMobile");
+    if (isInArgentMobileAppBrowser()) {
```
Comment (fricoben) [inline components/UI/walletConnect.tsx:36]: Add comment here saying `// Filter connectors and remove duplicates`

**Review by fricoben — CHANGES_REQUESTED** [review]
Good PR ! It works, just a few things and we're good

File: .env.test (hunk 1 above)
Comment (JoE11-y) [reply .env.test]: okay

File: components/UI/walletConnect.tsx (hunk 2 above)
Comment (JoE11-y) [reply components/UI/walletConnect.tsx]: it does.

File: .env.test [hunk 5]
```diff
@@ -2,3 +2,5 @@ NEXT_PUBLIC_VERIFIER_CONTRACT=0x057c942544063c3aea6ea6c37009cc9d1beacd750cb68015
 NEXT_PUBLIC_DEPRECATED_VERIFIER_CONTRACT=0x019e5204152a72891bf8cd0bed8f03593fdb29ceacd14fca587be5d9fcf87c0e
 NEXT_PUBLIC_OLD_VERIFIER_CONTRACT=0x4d546c8d60cfd591557ac0613be5ceeb0ea6f797e7d11c0b5160d145fa3089f
 NEXT_PUBLIC_VERIFIER_POP_CONTRACT=0x03528caf090179e337931ee669a5b0214041e1bae30d460ff07d2cea2c7a9106
+NEXT_PUBLIC_RPC_URL=https://starknet-sepolia.infura.io/v3
```
Comment (fricoben) [inline .env.test]: Change the RPC again pls `NEXT_PUBLIC_RPC_URL=https://sepolia.rpc.starknet.id`

**Review by fricoben — CHANGES_REQUESTED** [review]
Change the RPC again and we're good

File: .env.test (hunk 5 above)
Comment (JoE11-y) [reply .env.test]: Alright missed that.

**Review by fricoben — APPROVED** [review]
lgtm

**Pull request merged**